R/ctwas_wrapper.R

```diff
@@ -14,47 +14,132 @@ get_ctwas_meta_data <- function(ld_meta_data_file, regions_table) {
+        # adjust susie weights for selected number of weight variants 
+        if (colnames(post_qc_twas_data[[gene]][["weights_qced"]][[context]][["scaled_weights"]]) == "susie_weights"){
+          gene_data <- find_data(twas_weights_data, c(2, gene))
+          ld_variants <- colnames(post_qc_twas_data[[gene]][["LD"]])
+          pre_qc_variants <- gene_data[[context]][["variant_names"]]
+          qced_original_variants <- allele_qc(pre_qc_variants, ld_variants, pre_qc_variants, match_min_prop = 0)
+          idx_in_original_variants <- match(selected_variant_list[[context]], 
+                paste0("chr", qced_original_variants$qc_summary$variants_id_qced))
+          gene_data[[context]]$susie_weights_intermediate$mu <- gene_data[[context]]$susie_weights_intermediate$mu[, idx_in_original_variants, drop=FALSE]
+          gene_data[[context]]$susie_weights_intermediate$lbf_variable <- gene_data[[context]]$susie_weights_intermediate$lbf_variable[,idx_in_original_variants, drop=FALSE]
+          gene_data[[context]]$susie_weights_intermediate$X_column_scale_factors <- gene_data[[context]]$susie_weights_intermediate$X_column_scale_factors[idx_in_original_variants]
+          gene_data[[context]][["variant_names"]] <- selected_variant_list[[context]]
+          gene_data[[context]][["model_weights"]] <- postqc_scaled_weight
+          adjusted_susie_weights <- adjust_susie_weights(gene_data[[context]],
+              keep_variants = gene_data[[context]][["variant_names"]], allele_qc = TRUE,
+              variable_name_obj = c("variant_names"),
+              susie_obj = c("susie_weights_intermediate"),
+              twas_weights_table = c("model_weights"), ld_variants, match_min_prop = 0.001
+          )
+          postqc_scaled_weight <- matrix(adjusted_susie_weights$adjusted_susie_weights, ncol=1)
+          rownames(postqc_scaled_weight) <- paste0("chr", adjusted_susie_weights$remained_variants_ids)
+        }
+        colnames(postqc_scaled_weight) <- "weight"
+        context_variants <- selected_variant_list[[context]]
+        context_range <- sapply(context_variants, function(variant) as.integer(strsplit(variant, "\\:")[[1]][2]))
+
+        weights_list[[paste0(gene, "|", data_type, "_", context)]] <- list(
+          chrom = chrom,
+          p0 = min(context_range),
+          p1 = max(context_range),
+          wgt = postqc_scaled_weight,
+          #R_wgt = post_qc_twas_data[[gene]]$LD[context_variants, context_variants, drop = FALSE], # ld_list$combined_LD_matrix[context_variants, context_variants],
+          molecular_id = gene,
+          weight_name = paste0(data_type, "_", context),
+          type = data_type,
+          context = context,
+          n_wgt = length(context_variants)
+        )
+      }
+    }
+    # weights_list <- compute_weight_LD_from_ref(weights_list,
```

Comment:
As i said we don't know why we want to compute weight LD at this point. The input you prepare should not have any LD information included. It will be dealt with in cTWAS codes anyways, right?